enterprise/backend/src/metabase_enterprise/dependencies/api.clj

```diff
@@ -276,7 +279,33 @@
                                                         :exclude     [:and
                                                                       [:= active-column true]
                                                                       [:= visibility-type-column nil]]
-                                                        (:only :all) nil)]}]])))))
+                                                        (:only :all) nil)]}]])
+
+                     ;; Segment with table permissions and archived filtering
+                     :model/Segment
+                     (let [archived-column (keyword (name table-name) "archived")
```

Comment:
Not your fault, but I can't let it pass without expressing it - I find it deeply distressing that this generic, high-level API "fetch these entities and show them" has such a gory-details query embedded for each type. There's zero decoupling here and it makes me sad and anxious.

As I say, nothing for you to fix, though.